### ISSUE TITLE
Support manage ApplicationSet using Argo CD UI/API/CLI

### ISSUE THREAD

**Issue by alexmt** [argoproj/argo-cd#7352]
# Summary

Users should be able to manage ApplicationSet instances using Argo CD UI/CLI/API same as Application objects. 

# Motivation

In order to improve the experience users need the ability to manage ApplicationSet instances imperatively. 

# Proposal

TODO: create sub-tickets with detailed descriptions

**Comment by jgwest** [comment]
Back in May, I wrote up and presented the [the backend work required](https://github.com/argoproj-labs/applicationset/issues/244) for this.

**Comment by chetan-rns** [comment]
@alexmt @jgwest @wtam2018  I'd be happy to help with the integration of Appsets in the API and CLI. I think we should break this down into multiple smaller issues.

**Comment by machine3** [comment]
Are all UI/API/CLI supported?

**Comment by crenshaw-dev** [comment]
Good catch, @machine3! Just the API and CLI for now.

**Comment by machine3** [comment]
@crenshaw-dev  Why is it not supported to edit applicationset in api ?

**Comment by xmath279** [comment]
I've tried the RC version with this feature (CLI) and I was wondering if there were any plans to add some of the subcommands supported by the `argocd app` command. The current implementation (by https://github.com/argoproj/argo-cd/pull/9584) only supports `create`, `delete`, `get` and `list`. 

I have a use case where it would be useful to set parameters during deployment in a CI/CD for applicationsets. I currently use `kubectl patch` for that, but it has the disadvantage of requiring all parameters to be specified when overriding a value.

**Comment by nmohoric** [comment]
> Good catch, @machine3! Just the API and CLI for now.

As far as I can tell the rest API is not exposed (ie /api/v1/applicationsets) as this returns a 404 despite the swagger docs saying it should exist. Am I correct that only a grpc endpoint exists for this right now, or is there something wrong on my end?

**Comment by matthewdaltonfanduel** [comment]
Status on this? 2.5 is released with no mention of this as a feature or update on the roadmap

**Comment by crenshaw-dev** [comment]
afaik there is no in-flight work for more complete UI/CLI support for ApplicationSets. I'll update here if someone starts working on it.

**Comment by crenshaw-dev** [comment]
Someone started working on it. :-) https://github.com/argoproj/argo-cd/pull/13530

**Comment by stephanschielke** [comment]
> Someone started working on it. :-) #13530

Closed without reason :(

**Comment by wmcnamee-coreweave** [comment]
is there any progress on this? Even without the ability to edit anything, being able to _see_ the deployed applicationsets would be very nice.

**Comment by reedjosh** [comment]
+1

**Comment by sandeep-ongit** [comment]
> @wmcnamee-coreweave I think the effort is paused for the moment.
> 
> One workaround is to deploy the ApplicationSet via an App so that you can see the AppSet and child resources via the App UI.

Yes this is the current workaround I use until this new feature is rolled out .

**+1**

**Comment by thiagorider** [comment]
+1

**Comment by bygui86** [comment]
@crenshaw-dev may I ask you why the effort is paused?
thanks a lot in advance for your answer

**Comment by zswanson** [comment]
For those checking it's now tracked here https://github.com/argoproj/argo-cd/pull/16689
Seems like Regina is still working on it.

**Comment by PKatBK** [comment]
+1

**Comment by lieberlois** [comment]
+1

**Comment by kimminw00** [comment]
+1

**Comment by parthghetia-rh** [comment]
+1

**Comment by shingo-toma-rakuten** [comment]
+1

**Comment by ajaen4** [comment]
+1

**Comment by j-bolivar-lt** [comment]
+1

**Comment by docbyte86** [comment]
+1

**Comment by FarnazBGH** [comment]
+1

**Comment by LS80** [comment]
Commenting with a +1 just spams everyone subscribed to the thread. Consider reacting with a 👍 on the issue post instead?

**Comment by crenshaw-dev** [comment]
This feature has been merged and will be part of 3.5!
